src/component/TurnipCalculator.spec.tsx

```diff
@@ -180,4 +180,118 @@ describe('TurnipCalculator component', () => {
     });
     expect(handler).toHaveBeenLastCalledWith(1100);
   });
+
+  test("should not show the error alert if user doesn't do anything", () => {
+    const handler = jest.fn();
+
+    const { queryByTestId } = render(
+      <TurnipCalculator onTipCalculated={handler} />,
+    );
+
+    const errorAlert = queryByTestId('ErrorAlert');
+    expect(errorAlert).toBeNull();
+  });
+
+  test('should show the error alert if user types zero on the price field', () => {
+    const { getByRole, queryByTestId } = render(<TurnipCalculator />);
+
+    const priceField = getByRole('textbox', {
+      name: 'price',
+    });
+
+    fireEvent.change(priceField, {
+      target: {
+        value: 20,
+      },
+    });
+
+    fireEvent.change(priceField, {
+      target: {
+        value: 0,
+      },
+    });
+
+    const errorAlert = queryByTestId('ErrorAlert');
+    expect(errorAlert).toBeTruthy();
+  });
+
+  test('should show the error alert if user types zero on the turnips field', () => {
+    const { getByRole, queryAllByTestId } = render(<TurnipCalculator />);
+
+    const turnipsField = getByRole('textbox', {
+      name: 'turnips',
+    });
+
+    fireEvent.change(turnipsField, {
+      target: {
+        value: 0,
+      },
+    });
+
+    const errorAlert = queryAllByTestId('ErrorAlert');
+    expect(errorAlert).toBeTruthy();
+  });
+
+  test('should not show the error alert if user types a non-zero value on the price field', () => {
+    const { getByRole, queryByTestId } = render(<TurnipCalculator />);
+
+    const priceField = getByRole('textbox', {
+      name: 'price',
+    });
+
+    fireEvent.change(priceField, {
+      target: {
+        value: 20,
+      },
+    });
+
+    const errorAlert = queryByTestId('ErrorAlert');
+    expect(errorAlert).toBeNull();
+  });
+
+  test('should not show the error alert if user types a non-zero value on the turnips field', () => {
+    const { getByRole, queryByTestId } = render(<TurnipCalculator />);
+
+    const priceField = getByRole('textbox', {
+      name: 'price',
+    });
+
+    const turnipsField = getByRole('textbox', {
+      name: 'turnips',
+    });
+
+    fireEvent.change(priceField, {
+      target: {
+        value: 20,
+      },
+    });
+
+    fireEvent.change(turnipsField, {
+      target: {
+        value: 50,
+      },
+    });
+
+    const errorAlert = queryByTestId('ErrorAlert');
+    expect(errorAlert).toBeNull();
+  });
+
```

Comment:
Can you also add a test for showing two error alerts if user enters 0 on both price and turnips fields? You can use [`queryAllByTestId`](https://testing-library.com/docs/dom-testing-library/api-queries#queryallby) to accomplish this I think.

Reply:
Ok! Added on the next commit! Thanks for the feedback!